### DISCUSSION TITLE
windows usbipd-win wsl2支持

### DISCUSSION THREAD

**Discussion by guantongpeng** [adoyle-h/my-development-tools#8] [Ideas]
首先感谢您提供的这个很棒的仓库。
关于windows usbipd-win wsl2支持上可能存在一些描述不准确和值得注意的问题，具体如下：

1. **usbipd支持wsl2 adb**，但是可能`adb devices`不生效，此时需要`adb kill-server`，并使用sudo权限重新开启adb`sudo adb devices`，就可以看见adb设备了，参考[官方仓库issue](https://github.com/dorssel/usbipd-win/issues/60)。
2. 使用usbipd**对于wsl2版本有要求**，使用前可能需要升级wsl2版本`wsl --update`，参考[官方Wiki](https://github.com/dorssel/usbipd-win/wiki/WSL-support)
3. **usbipd还支持局域网内设备访问**

**Comment by adoyle-h** [comment]
感谢你的指正，稍后我会更新到文本里。确实如你所说。先 `adb kill-server` 然后用 sudo 权限开启 adb server 是关键。

另外我还踩了几个坑，在这也分享下。
1. windows 防火墙默认会阻止 WSL 访问宿主机。需要在 Windows Defender 防火墙里添加**入站规则**。
2. 当执行 `adb shell` 看到报错 `This adb server's $ADB_VENDOR_KEYS is not set` 时，去看一眼手机，要点授权后才能访问。需要 `adb kill-server` 重来一遍。
3. 启用 adb 之前，先在 WSL 里用 lsusb 看看是否识别了新的 usb 接口。
4. 如果以上都确保没问题，还是没法连通宿主机。可以排查电脑里安装的有关网络管理、监控、VPN 等应用。可能是它搞得鬼。之前我装了一个网络监控工具把我电脑的网络流量给劫持了，导致修改防火墙规则也不起作用。
